pkgs/development/compilers/llvm/11/default.nix

```diff
@@ -248,9 +249,12 @@ let
 
     compiler-rt-libc = callPackage ./compiler-rt {
       inherit llvm_meta;
-      stdenv = if (stdenv.hostPlatform.useLLVM or false) || (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) || (stdenv.hostPlatform.isRiscV && stdenv.hostPlatform.is32bit)
-               then overrideCC stdenv buildLlvmTools.clangNoCompilerRtWithLibc
-               else stdenv;
+      # Build compiler-rt without CF to prevent an infinite recursion when cross-compiling to Darwin.
+      stdenv = if stdenv.hostPlatform.isDarwin && (stdenv.buildPlatform != stdenv.hostPlatform)
+        then overrideCC darwin.stdenvNoCF buildLlvmTools.clangNoCompilerRtWithLibc
+        else if stdenv.hostPlatform.useLLVM or false || stdenv.isDarwin || (stdenv.hostPlatform.isRiscV && stdenv.hostPlatform.is32bit)
```

Comment:
It still does. Note that line 255 includes `stdenv.isDarwin`. This is mirroring the logic from LLVM 16.

Reply:
The reason for having these two cases is because building without CF prevents building the sanitizers (because ASAN requires CoreFoundation). Ideally, the native builds would have sanitizers. This actually fixes aarch64-darwin to build sanitizers since it’s not building them currently.

Reply:
So it is intentional for native aarch64-darwin to not match the overrideCC case then. Good.